pkg/ingester/flush.go

```diff
@@ -441,9 +441,11 @@ func (i *Ingester) flushChunks(ctx context.Context, fp model.Fingerprint, labelP
 		)
 
 		// encodeChunk mutates the chunk so we must pass by reference
+		chunkMtx.Lock()
 		if err := i.encodeChunk(ctx, &ch, c); err != nil {
 			return err
```

Comment:
should you unlock here too ?

Reply:
Yep, on line 448 below. I can wrap this in a func if you prefer to use defer

Reply:
No that's fine